### ISSUE TITLE
nuget.exe restore doesn't produce .props and .targets files for .nuproj projects (regression in v3.4.3.855)

### ISSUE THREAD

**Issue by AArnott** [NuGet/Home#2711]
Repro steps:

```
git clone https://github.com/AArnott/PCLCrypto.git
cd PCLCrypto
git checkout invalidPortableFx
nuget restore src
```

**Expected** (NuGet 3.3 behavior)

NuGet restore places adds a src\PCLCrypto.NuGet\PCLCrypto.NuGet.nuget.props and .targets file in my source tree.

**Actual** (NuGet 3.4.3 behavior)

No .props and .targets files are added to the src\PCLCrypto.NuGet directory.

**Comment by emgarten** [comment]
Is restore succeeding for you? I get an error saying `Invalid portable frameworks.` when restoring the `src` folder.

**Comment by AArnott** [comment]
Sorry, I left out a step to check out a branch where I've fixed that issue.

**Comment by AArnott** [comment]
That issue is also new in NuGet 3.4. 3.3 was wiling to tolerate 'net-cf' but with 3.4 I had to fix that. So the update to the repro steps I just made switches to the branch where I've updated my package dependencies to remove the net-cf reference.

**Comment by emgarten** [comment]
Thanks, I was able to repro this. Nuproj isn't a known extension type in NuGet right now so it gets skipped. This was added in 3.4.0 along with support for restoring .Xproj which does not support msbuild targets. 

I think the fix here is to generate targets for all project types except xproj, instead of only generating targets for known msbuild types. And nuproj should be added to that list as well.

**Comment by yishaigalatzer** [comment]
Or we can add a project capability

**Comment by emgarten** [comment]
Fixed in 3.4.4

NuGet/NuGet.Client@f4b3fdc2512fe31cfe147de3195ca70cbcd2a78c

**Comment by AArnott** [comment]
Thank you, @emgarten!

**Comment by joelverhagen** [comment]
@AArnott and I both tried this offline and we're seeing a different error now:

<pre>
MSBuild auto-detection: using msbuild version '15.0' from 'C:\Program Files (x86)\MSBuild\15.0\bin'.
Restoring packages for D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto.Tests.Desktop\project.json...
Restoring packages for D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto.WindowsPhone\project.json...
Restoring packages for D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto.iOS\project.json...
Restoring packages for D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto\project.json...
<b><i>- truncated -</i></b>
Committing restore...
Generating MSBuild file PCLCrypto.Desktop.nuget.targets.
D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto.Desktop\PCLCrypto.Desktop.csproj
Restore completed in 458ms.
Restoring packages for D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto.Tests.UWP\project.json...
Committing restore...
Committing restore...
Generating MSBuild file PCLCrypto.Tests.Desktop.nuget.targets.
Generating MSBuild file PCLCrypto.Tests.WinRT.nuget.targets.
Committing restore...
Generating MSBuild file PCLCrypto.Tests.Desktop.nuget.props.
Generating MSBuild file PCLCrypto.Tests.WinRT.nuget.props.
Generating MSBuild file PCLCrypto.iOS-Unified.nuget.targets.
D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto.Tests.Desktop\PCLCrypto.Tests.Desktop.csproj
Restore completed in 594ms.
D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto.Tests.WinRT\PCLCrypto.Tests.WinRT.csproj
Restore completed in 595ms.
Restoring packages for D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto.Tests.WinRT.Phone\project.json...
Restoring packages for D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto.NuGet\project.json...
D:\trash\2016-05-05\PCLCrypto\src\PCLCrypto.iOS-Unified\PCLCrypto.iOS-Unified.csproj
Restore completed in 173ms.
<b>Invalid portable frameworks.</b>
</pre>

The following files are generated during this process:

```
Removing src/PCLCrypto.Android/PCLCrypto.Android.nuget.targets
Removing src/PCLCrypto.Android/project.lock.json
Removing src/PCLCrypto.Desktop/PCLCrypto.Desktop.nuget.targets
Removing src/PCLCrypto.Desktop/project.lock.json
Removing src/PCLCrypto.Tests.Desktop/PCLCrypto.Tests.Desktop.nuget.props
Removing src/PCLCrypto.Tests.Desktop/PCLCrypto.Tests.Desktop.nuget.targets
Removing src/PCLCrypto.Tests.Desktop/project.lock.json
Removing src/PCLCrypto.Tests.WinRT/PCLCrypto.Tests.WinRT.nuget.props
Removing src/PCLCrypto.Tests.WinRT/PCLCrypto.Tests.WinRT.nuget.targets
Removing src/PCLCrypto.Tests.WinRT/project.lock.json
Removing src/PCLCrypto.WinRT/PCLCrypto.WinRT.nuget.targets
Removing src/PCLCrypto.WinRT/project.lock.json
Removing src/PCLCrypto.WindowsPhone/PCLCrypto.WindowsPhone.nuget.targets
Removing src/PCLCrypto.WindowsPhone/project.lock.json
Removing src/PCLCrypto.iOS-Unified/PCLCrypto.iOS-Unified.nuget.targets
Removing src/PCLCrypto.iOS-Unified/project.lock.json
Removing src/PCLCrypto.iOS/PCLCrypto.iOS.nuget.targets
Removing src/PCLCrypto.iOS/project.lock.json
Removing src/PCLCrypto/PCLCrypto.nuget.targets
Removing src/PCLCrypto/project.lock.json
```

**Comment by joelverhagen** [comment]
Looks like your PCLCrypto.NuGet project.json is depending on ReadOnlySourceTree 0.1.24-beta instead of 0.1.36-beta (which [seems to have been fixed](https://github.com/AArnott/ReadOnlySourceTree/commit/cc25ea52c97ecc63f8ab20c056f20fe93758b7c9#diff-f7d5587c1307d05f77a447fb9776d44a) to remove that invalid TFM).

Does bumping that dependency version (along with using the latest nuget.exe) unblock you?

**Comment by joelverhagen** [comment]
To clarify, I am talking about this line:
https://github.com/AArnott/PCLCrypto/blob/ef3645121224d4f9f85e9594f04ac560b63375a7/src/PCLCrypto.NuGet/project.json#L12

**Comment by AArnott** [comment]
Yes, fixing that allowed use of 3.4.4, which does generate the .props and .targets files. 
It's curious that 3.4.4 stumbled on the invalid portable frameworks error when 3.4.3 didn't though. Should I open another issue to improve the error message to specify where it found an invalid framework to make it quicker to find what packages need to be updated? Who else is hitting this besides me?

Thanks for your help.

**Comment by joelverhagen** [comment]
> It's curious that 3.4.4 stumbled on the invalid portable frameworks error when 3.4.3 didn't though.

3.4.3 had the same framework validations that 3.4.4 has. 3.4.3 did not get to the point that 3.4.4 did because it does not recognized the `.nuproj` project type.

**Comment by KirillOsenkov** [comment]
Is there a workaround for not discovering .nuproj if I'm stuck on 3.4.3?

Or will 3.4.4 be released soon enough?

**Comment by yishaigalatzer** [comment]
Very soon

Get Outlook for Androidhttps://aka.ms/ghei36

On Fri, May 20, 2016 at 11:09 PM -0700, "Kirill Osenkov" <notifications@github.com<mailto:notifications@github.com>> wrote:

Is there a workaround for not discovering .nuproj if I'm stuck on 3.4.3?

Or will 3.4.4 be released soon enough?

##
